packages/swapper/src/swappers/PortalsSwapper/utils/fetchPortalsTradeOrder.ts

```diff
@@ -0,0 +1,88 @@
+import axios from 'axios'
+
+// non-exhaustive
+type PortalsTradeOrderParams = {
+  sender: string
+  inputToken: string
+  inputAmount: string
+  outputToken: string
+  slippageTolerancePercentage?: number
+  // Technically optional, but we always want to use an affiliate addy
+  partner: string
+  feePercentage?: number
+  // Technically optional, but we want to explicitly specify validate
+  validate: boolean
+}
+
+type PortalsTradeOrderResponse = {
+  context: {
+    orderId: string
+    inputToken: string
+    inputAmount: string
+    inputAmountUsd: number
+    outputToken: string
+    outputAmount: string
+    outputAmountUsd: number
+    minOutputAmount: string
+    minOutputAmountUsd: number
+    slippageTolerancePercentage: number
+    target: string
+    partner: string
+    sender: string
+    recipient: string
+    value: string
+    route: string[]
+    routeHash: string
+    steps: string[]
+    gasLimit?: string
+    feeToken?: string
+    feeAmount?: string
+    feeAmountUsd?: number
+  }
+  tx: {
+    to: string
+    from: string
+    data: string
+    value: string
+    gasLimit: string
+  }
+}
+export const fetchPortalsTradeOrder = async ({
+  sender,
+  inputToken,
+  inputAmount,
+  outputToken,
+  slippageTolerancePercentage,
+  partner,
+  feePercentage,
+  validate,
+}: PortalsTradeOrderParams): Promise<PortalsTradeOrderResponse> => {
+  const url = 'https://api.portals.fi/v2/portal'
```

Comment:
suggestion: should be an environment variable so we can hotswap without releasing

Reply:
Not going to tackle since I couldn't find any env references in `packages/swapper` and we may not support env injection gracefully following the swapper packages migration

Reply:
Using `SwapperConfig`, but I don't mind tbh